src/ui/components/shared/CommentTool.tsx

```diff
@@ -83,81 +83,87 @@ function CommentTool({
   const { userId } = useGetUserId();
   const captionNode = useRef<HTMLDivElement | null>(null);
 
-  const addListeners = () => {
+  // Re-register the listener on every update to prevent the props used by the handler functions from being stale.
+  useEffect(() => {
     const videoNode = document.getElementById("graphics");
-
     if (videoNode) {
+      const onMouseDown = (evt: MouseEvent) => {
```

Comment:
This looks like a lot of changes but the bulk of it is moving event handlers inside of the effect body. We should be doing this as a general rule, to avoid potential stale closure issues.